flink-streaming-java/src/test/java/org/apache/flink/streaming/runtime/tasks/StreamTaskTest.java

```diff
@@ -180,6 +180,25 @@
 	@Rule
 	public final Timeout timeoutPerTest = Timeout.seconds(30);
 
+	/**
+	 * This test checks the async exceptions handling wraps the message and cause as an AsynchronousException
+	 * and propagates this to the environment.
+	 */
+	@Test
+	public void exceptionReporting()  {
+		Environment e = mock(Environment.class);
+		RuntimeException expectedException = new RuntimeException("RUNTIME EXCEPTION");
+
+		SourceStreamTask sut = new SourceStreamTask(e);
+		sut.handleAsyncException("EXPECTED_ERROR", expectedException);
+
+		ArgumentCaptor<AsynchronousException> c = ArgumentCaptor.forClass(AsynchronousException.class);
+		verify(e).failExternally(c.capture());
+		assertEquals(c.getValue().getMessage(), "EXPECTED_ERROR");
+		assertEquals(c.getValue().getCause(), expectedException);
+		assertEquals(expectedException, "AsynchronousException{EXPECTED_ERROR, caused by RUNTIME EXCEPTION}");
```

Comment:
Instead of mocking here via Mockito I would suggest to use `MockEnvironment` and call `#setExpectedExternalFailureCause` and `#getActualExternalFailureCause`.

Reply:
done